src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
this code snippet also could be optimized and rewritten to use 2 filters one after another.
It would be good to use better namings to get a clear understanding of what every line of code does.
```suggestion
  const women = people
    .filter((person) => person.sex === 'f')
    .filter((person) => withChildren ? people.find((child) => child.mother === person.name) : true);
```